F 03

vladn255 opened the pull request.

vladn255 committed:
F-03. City choice

- location-form, text-inputs refactoring;
- leaflet to yandex-maps migration;
- picking city implemented;
WIP: setting pickpoints

vladn255 committed:
F-03. Pickpoint pick feature

- implemented Placemarks at yandex-maps;
- active marker centers the map.

vladn255 committed:
F-03 Bugfixes, refactor

- implemented fetching model categories.
- few minor refactors at model-form and model-form-item

AlexsandrLitkevich commented on src/components/autocomplete-text-inputs/autocomplete-text-inputs.jsx:
```diff
@@ -0,0 +1,114 @@
+
+
+const AutocompleteTextInputs = ({ values: { city, pickpoint }, citiesData, pickpointData, setLocationDataValue, resetLocationData }) => {
+    const inputCityRef = useRef('');
```
лучше начальное значение null

AlexsandrLitkevich commented on src/components/autocomplete-text-inputs/autocomplete-text-inputs.jsx:
```diff
@@ -0,0 +1,114 @@
+
+        getFilteredSuggestions(citiesData, evt.target.value, setCities, setIsListVisible)
+
+        if (evt.target.value.length === 0) {
```
если написать так !evt.target.value.length результат тот и читается лучше

AlexsandrLitkevich commented on src/components/location-form/location-form.jsx:
```diff
@@ -3,71 +3,109 @@ import { useDispatch, useSelector } from "react-redux";
+            })
+        }
+
+        if (citiesData.filter((city) => city.name === locationData.city)) {
```
тут условие подумай как можно лучше. или вообще убери if или ищи совпадения при срабатывание эффекта

AlexsandrLitkevich replied:
или просто функцию напиши и не надо if

AlexsandrLitkevich commented on src/components/location-form/location-form.jsx:
```diff
@@ -3,71 +3,109 @@ import { useDispatch, useSelector } from "react-redux";
+        if (locationValue.name === Labels.CITY) {
+            const filteredCity = citiesData.slice().filter((item) => item.value === locationValue.value)
 
+            if (filteredCity.length !== 0) {
```
достаточно просто написать filteredCity.length и если > 0 то true если равно 0 то false

AlexsandrLitkevich commented on src/components/map-component/map-component.jsx:
```diff
@@ -0,0 +1,102 @@
+const MapComponent = ({ activeCity, activeMarker, markersData }) => {
+
+    const [savedYmaps, setSavedYmaps] = useState(null)
+    const [zoom, setZoom] = useState(defaultlMark.zoom)
```
используй диструктуризацию

AlexsandrLitkevich commented on src/components/map-component/map-component.jsx:
```diff
@@ -0,0 +1,102 @@
+
+    const [savedYmaps, setSavedYmaps] = useState(null)
+    const [zoom, setZoom] = useState(defaultlMark.zoom)
+    const [activeCityCoords, setActiveCityCoords] = useState(defaultlMark.center)
```
используй диструктуризацию

AlexsandrLitkevich commented on src/components/map-component/map-old.jsx:
```diff
@@ -0,0 +1,75 @@
+const Map = ({ markers, activeMarker, activeCity }) => {
+    const mapRef = useRef();
+    const customIcon = leaflet.icon({
+        // eslint-disable-next-line no-undef
```
подумай про настройки линта.добавь в список разрешённых глобальных переменных https://eslint.org/docs/user-guide/configuring#specifying-globals

AlexsandrLitkevich commented on src/components/model-form-item/model-form-item.jsx:
```diff
@@ -7,14 +7,18 @@ const ModelFormItem = ({ name, priceMin, priceMax, imgSrc, isActive, setModelVal
-        isValid = false;
-    } else if ((typeof imgSrc === 'string')) {
-        if (!(imgSrc.includes('jpg') || imgSrc.includes('jpeg') || imgSrc.includes('png'))) {
+        if (!(typeof imgSrc === 'string')) {
```
почему не написать typeof imgSrc !== 'string' ?????

AlexsandrLitkevich commented on src/components/model-form-item/model-form-item.jsx:
```diff
@@ -7,14 +7,18 @@ const ModelFormItem = ({ name, priceMin, priceMax, imgSrc, isActive, setModelVal
-        if (!(imgSrc.includes('jpg') || imgSrc.includes('jpeg') || imgSrc.includes('png'))) {
+        if (!(typeof imgSrc === 'string')) {
             isValid = false;
+        } else if ((typeof imgSrc === 'string')) {
```
зачем двойные скобки?

AlexsandrLitkevich commented on src/components/model-form/model-form.jsx:
```diff
@@ -44,12 +36,23 @@ const ModelForm = () => {
+    const [categories, setCategories] = useState(MODEL_FORM_RADIO_DEFAULT);
 
-    if (models.length === 0) {
+    if (models !== undefined && models.length === 0) {
```
тут напиши models?.length === 0 или !models?.length 
? - https://javascript.info/optional-chaining

AlexsandrLitkevich commented on src/components/model-form/model-form.jsx:
```diff
@@ -85,9 +86,9 @@ const ModelForm = () => {
                         <legend className="visually-hidden">Форма выбора модели</legend>
                         <ul className="model-form__list">
-                            {ModelFormRadios.map(({ name, form, label }) =>
+                            {categories.map(({ name, form }) =>
```
name это стабильное и уникальное значение??? Оно подходит для KEY

vladn255 committed:
F-03. Bugfixes

- refactored conditions at location and model forms;
- added entry at eslintrc;
- pickpoint fetching refactoring;
